src/lib/esnext.collection.d.ts

```diff
@@ -9,3 +9,49 @@ interface MapConstructor {
         keySelector: (item: T, index: number) => K,
     ): Map<K, T[]>;
 }
+
+interface SetLike<T> {

```

Comment:
At first glance, this looks a lot like `ReadonlySet`. I know that these methods only need a subset of functionality, but it might be worth using that instead.

Reply:
You need the `keys` method, so I don't think that works unless that gets added ReadonlySet as well. (And users shouldn't have to implement `forEach` for their own set-likes, though that isn't as big a deal since that probably doesn't come up as much.)

Reply:
Wait, no, `keys` is already there, just in a different file, whoops.

Still, I think it would be kind of annoying not to be able to do `set.union({ keys(){ return array.keys() }, size: array.size, has: () => {} })` or similar. Not the end of the world, but I'd prefer to keep `SetLike` unless there's a reason to switch to `ReadonlySet`.